src/main/java/org/oscii/detokenize/Labeler.java

```diff
@@ -0,0 +1,74 @@
+package org.oscii.detokenize;
+
+import com.google.common.collect.Iterators;
+import com.google.common.collect.PeekingIterator;
+
+import java.text.Normalizer;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Infer label sequence from a preprocessor applied to a string.
+ * <p>
+ * TODO(denero) Infer capitalization and replacement
+ * <p>
+ * Note: The logic in this class could be removed if we exposed the CoreLabel
+ * sequence of the pre-processor, but since we may move away from CoreNLP, lets
+ * keep it for now.
+ */
+public class Labeler {
+    public List<TokenLabel> getLabels(String sentence, List<String> tokens) throws LabelException {
+        sentence = Normalizer.normalize(sentence, Normalizer.Form.NFC);
```

Comment:
I am not sure if we should apply further normalization here, as this will introduce surface forms that are different from what the MT systems will produce. There is some normalization applied during preprocessing (com.lilt.process.CharacterCleanup). Is this normalization step applied both during training of the detok models and actual detokenization if new input? If yes, then it should be fine.